summary/summary.go

```diff
@@ -55,12 +55,23 @@ var _ Summarizer[*types.CGMPeriods, *types.GlucoseBucket, types.CGMPeriods, type
 var _ Summarizer[*types.BGMPeriods, *types.GlucoseBucket, types.BGMPeriods, types.GlucoseBucket] = &GlucoseSummarizer[*types.BGMPeriods, *types.GlucoseBucket, types.BGMPeriods, types.GlucoseBucket]{}
 var _ Summarizer[*types.ContinuousPeriods, *types.ContinuousBucket, types.ContinuousPeriods, types.ContinuousBucket] = &GlucoseSummarizer[*types.ContinuousPeriods, *types.ContinuousBucket, types.ContinuousPeriods, types.ContinuousBucket]{}
 
-func CreateSelfMonitoredGlucoseDatum() data.Datum {
-	return selfmonitored.New()
+func CreateSelfMonitoredGlucoseDatum(typ string) (data.Datum, error) {
```

Comment:
nit: I'd prefer not to pass arguments that aren't used. It's not implementing an interface.

Various linters will complain about this as well.

If they need to be there for fetcher.NewDefaultCursor then we can name the variable `_` to quiet the linters, and also let the reader know that the parameter is intentionally ignored.

Reply:
It is implementing an interface, but golang makes this rather difficult to see

Reply:
```
type DatumCreator func(typ string) (data.Datum, error)
```